### ISSUE TITLE
Backports to <=8.4 require manual work due to zeebe sub-folder migration on main

### ISSUE THREAD

**Issue by megglos** [camunda/zeebe#17218]
**Description**

Since the merge of https://github.com/camunda/zeebe/pull/16522 which is only present on `main`, backports fail when new files were introduced in the sub-path. These "soft" conflicts always require manual work.

Examples of backport failures because of the sub-folder mismatch between `main` and <`stable/8.4` just because of added files:

https://github.com/camunda/zeebe/pull/17124#issuecomment-2025519529 (new classes in exporter)
https://github.com/camunda/zeebe/pull/16684#issuecomment-1978526087 (New Test class)

**Comment by Zelldon** [comment]
Discussed with @romansmirnov :

At the start we agreed on that we only do it on main, and it works for most changes. As it only impacts new files/additions we would argue that was out of scope before, so we would like not do it. If it is a pain for the team the respective team has to handle it.

Let me know what you think.

**Comment by megglos** [comment]
Followed-up with Roman and agreed that we accept the manual effort for backports that introduce new files for now (it should be only once to 8.4 and from there be automatically backportable again). The alternative to change all stable branches will likely not be a one time thing as other moves are likely to happen a couple of times more as we progress with the streamlining.
